### PR TITLE
Sync old chain segment after cold start with SYNC_MODE=peer 

### PR DESCRIPTION
**Pull request by platfowner**
Change summary:
- Sync old chain segment after cold start with SYNC_MODE=peer 
- Re-assign sync peer when pre-check fails
- Upgrade data protocol version to 1.1.0 (DATA_PROTOCOL_VERSION)

Related issues:
- https://github.com/ainblockchain/ain-blockchain/issues/713

### CODE REVIEW

File: p2p/index.js [hunk 1]
```diff
@@ -956,6 +995,126 @@ class P2pClient {
...
+      logger.info(
+          `[${LOG_HEADER}] Old chain segment of size ${segmentSize} ` +
+          `(${fromBlockNumber} ~ ${toBlockNumber}) from ${senderAddress} was written.`);
+    } 
+
+    const oldestBlockNumber = this.oldChainSyncInProgress.oldestBlockNumber;
+    if (oldestBlockNumber > 0) {
+      // Buffer time to avoid network resource abusing
+      await CommonUtil.sleep(NodeConfigs.OLD_CHAIN_SEGMENT_SLEEP_MS);
+      this.requestOldChainSegment();
```
Comment (liayoo) [inline p2p/index.js:1069]: question) do we not need to change the peer (force reset) if segmentSize === 0?

File: common/file-util.js [hunk 2]
```diff
@@ -157,18 +157,35 @@ class FileUtil {
...
         logger.debug(`[${LOG_HEADER}] blockFile (${blockFile}) does not exist`);
-        return blockPaths;
+        return blockPathList;
       }
     }
-    return blockPaths;
+    return blockPathList;
+  }
+
+  static getOldBlockPathList(chainPath, from, size) {
```
[the remark this answers is not in the document]
Comment (platfowner) [reply common/file-util.js:190]: Done, thanks!

File: p2p/index.js (hunk 1 above)
Comment (platfowner) [reply p2p/index.js]: Right, just updated, PTAL!

**Comment by platfowner** [comment]
All comments addressed, PTAL @liayoo

**Review by liayoo — APPROVED** [review]
LGTM, thanks!

**Comment by platfowner** [comment]
Thanks for the review! Now merging.. @liayoo

**Pull request merged**